tsdb/engine/tsm1/encoding.go

```diff
@@ -769,6 +979,56 @@ func (a StringValues) Filter(min, max int64) StringValues {
 	return a[:i]
 }
 
+// Include returns the subset values between min and max inclusive.
+func (a StringValues) Include(min, max int64) StringValues {
+	var i int
+	for j := 0; j < len(a); j++ {
+		if a[j].UnixNano() < min || a[j].UnixNano() > max {
+			continue
+		}
+
+		a[i] = a[j]
+		i++
+	}
+	return a[:i]
+}
+
+// Merge overlays b to top of a.  If two values conflict with
+// the same timestamp, b is used.  Both a and b must be sorted
+// in ascending order.
+func (a StringValues) Merge(b StringValues) StringValues {
+	if len(a) == 0 {
+		return b
+	}
+
+	if len(b) == 0 {
+		return a
+	}
+
+	var i, j int
+	for i < len(a) && j < len(b) {
+		if a[i].UnixNano() < b[j].UnixNano() {
+			i++
+		} else if a[i].UnixNano() > b[j].UnixNano() {
+			a[i], b[j] = b[j], a[i]
+			i++
+		} else {
+			a[i] = b[j]
+			i++
+			j++
+		}
+	}
+
+	if i >= len(a) {
+		if j+1 < len(b) && b[j].UnixNano() == b[j+1].UnixNano() {
+			j++
+		}
+		return append(a, b[j:]...)
+	}
+
+	return a
+}
```

Comment:
Yes.  I had the same thought.  Felt like I was duplicating way too much code here and in `FileStore`.